client/data/happiness-engineers/use-happiness-engineers-query.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * External dependencies
+ */
+import { useQuery } from 'react-query';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'calypso/lib/wp';
+
+const useHappinessEngineersQuery = () =>
+	useQuery( 'happinessEngineers', async () => await wpcom.undocumented().getHappinessEngineers(), {
```

Comment:
I was going to ask if this was documented somewhere, but I guess it's not? :D ... what's the story behind this API?

Reply:
The story behind this API goes back to 412-gh-Automattic/calypso-pre-oss and that happened 7 years ago ⌛ !

The idea initially was to have a client-side library that handles undocumented (private) APIs in wpcom.js. 

But because it was easy enough and low-effort to create new methods, it eventually became much more than that and was used for both private and public endpoints. I think it went out of control at some point 😉